dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/search/PresetsServlet.java

```diff
@@ -65,24 +86,65 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         String token = req.getHeader("Authorization");
         User user = Authentication.getInstance().getUsername(token);
         if (user == null) {
-            resp.sendError(401);
+            sendError(resp, 401, "Unauthorized access to this user.");
+            return;
         }
 
-        Map<String, String> presets = PresetsHandler.getPresets(user.getUsername());
-        if (presets == null) {
-            resp.sendError(500, "Could not retrieve presets.");
+        List<String> pathParts = sanitizedSubpathParts(req);
+
+        // the format must be /<username>
+        if (pathParts.size() != 1) {
+            sendError(resp, 400, "Illegal plugin request URI: wrong resource path size");
+            return;
+        }
+
+        String username = pathParts.get(0);
+
+        // if the specified user is not the requester, he must have admin privileges
```

Comment:
s/he/they/ (_", they must have admin privileges"_). Not important, but it's best to keep users gender-neutral.